code/modules/reagents/reagent_containers/food.dm

```diff
@@ -2,19 +2,35 @@
 /// Food.
 ////////////////////////////////////////////////////////////////////////////////
 /obj/item/reagent_containers/food
+	drop_sound = 'sound/items/drop/food.ogg'
 	flags = OPENCONTAINER
 	possible_transfer_amounts = null
 	volume = 50 //Sets the default container amount for all food items.
 	var/filling_color = "#FFFFFF" //Used by sandwiches
-	drop_sound = 'sound/items/drop/food.ogg'
-	var/is_liquid = FALSE // Handles whether you can eat with a fork & which eating sound is played
-
-/obj/item/reagent_containers/food/self_feed_message(var/mob/user)
-	to_chat(user, "<span class='notice'>You [is_liquid? "drink" : "eat"] \the [src].</span>")
+	var/trash = null
+	var/is_liquid = TRUE
 	
+/obj/item/reagent_containers/food/self_feed_message(var/mob/user)
+	to_chat(user, "<span class='notice'>You [is_liquid? "drink from" : "eat"] \the [src].</span>")
 
 /obj/item/reagent_containers/food/feed_sound(var/mob/user)
 	if(is_liquid)
 		playsound(user.loc, 'sound/items/drink.ogg', rand(10, 50), 1)
 	else
 		playsound(user.loc, 'sound/items/eatfood.ogg', rand(10, 50), 1)
+
+/obj/item/reagent_containers/food/proc/on_consume(var/mob/user = null, var/mob/target)
+	if(!reagents.total_volume)
+		if(!user)
+			user = target
+		user.drop_from_inventory(src)	//so trash actually stays in the active hand.
+		if(trash)
+			if(ispath(trash,/obj/item))
+				var/obj/item/TrashItem = new trash(user)
+				user.put_in_hands(TrashItem)
+			else if(istype(trash,/obj/item))
+				user.put_in_hands(trash)
```

Comment:
What is this even for? Just make sure they're all types.

Reply:
I have no idea. This is oldcode.